chess-AI/train.py

```diff
@@ -247,25 +231,56 @@ def main():
     if dataset_path:
         dataset = torch.load(dataset_path)
     else:
-        dataset = create_dataset(stocktrain_games, stocktrain_moves, stocktrain_value_approximator)
+        dataset = create_dataset(options.games, stocktrain_moves, stocktrain_value_approximator)
         # NOTE: Dataset should be given a more descriptive name, this is just temporary
         torch.save(dataset, './datasets/stockfish_data.pt')
 
     # Train using the stockfish dataset
-    train_on_dataset(dataset, nnet, learning_rate, momentum, weight_decay, stocktrain_epochs, batch_size, device,
-                     save_path=None, num_saved_models=num_saved_models, overwrite_save=overwrite_save)
+    train_on_dataset(dataset, nnet, options)
+
 
+def train_on_mcts(nnet, exploration, mcts_simulations, training_episodes, options):
     # Get MCTS object and parameters
     # TODO: Figure out how many times to perform self play (and better name for this variable)
-    mcts = Mcts(exploration)
+    mcts = Mcts(exploration, options.device)
 
     # Will iterate through the number of training episodes
-    for _ in range(mcts_amt):
+    for _ in range(training_episodes):
         mcts_moves = lambda board: mcts.get_tree_results(mcts_simulations, nnet, board, temperature=5)
 
-        dataset = create_dataset(mcts_games, mcts_moves)
-        train_on_dataset(dataset, nnet, learning_rate, momentum, weight_decay, mcts_epochs, batch_size, device,
-                         save_path=None, num_saved_models=num_saved_models, overwrite_save=overwrite_save)
+        dataset = create_dataset(options.games, mcts_moves)
+        train_on_dataset(dataset, nnet, options)
+
+
+def main():
+    """Main function that will be run when starting training
+    """
+
+    # Detect device to train on
+    device = torch.device('cuda:0' if torch.cuda.is_available() else 'cpu')
+    nnet = PlayNetwork().to(device=device)
+
+    # Will get the paths to load models and datasets from
+    model_path = None
+    dataset_path = None
+
+    num_saved_models = 5
+    overwrite = True
+
+    # Load in a model
+    load_model(nnet, model_path, num_saved_models)
+
+    # Train network using stockfish evaluations
+    stockfish_options = TrainOptions(learning_rate=0.1, momentum=0.9, weight_decay=0.001, epochs=10, batch_size=8, games=1000,
+                                     device=device, save_path=model_path, num_saved_models=num_saved_models, overwrite=overwrite)
+
+    train_on_stockfish(nnet, elo=1000, depth=3, dataset_path=dataset_path, options=stockfish_options)
+
+    # Train network using MCTS
+    mcts_options = TrainOptions(learning_rate=0.1, momentum=0.9, weight_decay=0.001, epochs=100, batch_size=8, games=100,
+                                device=device, save_path=model_path, num_saved_models=num_saved_models, overwrite=overwrite)
+
+    train_on_mcts(nnet, exploration=5, training_episodes=5, mcts_simulations=100, options=mcts_options)
```

Comment:
Can you pls make variables for all the parameters specified in the function call? It'll make it easier for @ryPattillo and I to update this code to handle either the dashboard/reading from json. You can initialize the variables with the values here, and we can use those for reference as the default for the hyperparams.json file we'll make